core/src/main/java/org/apache/iceberg/TableProperties.java

```diff
@@ -40,6 +40,13 @@ private TableProperties() {
    */
   public static final String FORMAT_VERSION = "format-version";
 
+  /**
+   * Reserved table property for UUID.
+   * <p>
+   * This reserved property is used to store the UUID of the table.
+   */
+  public static final String UUID = "uuid";
+
   /**
    * Reserved Iceberg table properties list.
    * <p>
```

Comment:
I think you might need to add the new constant `UUID` to this set of reserved properties just below this line.

Reply:
Good point. Looking more deeply, it isn't a reserved table properties. Will remove it from there.

Reply:
Also add a test for that.